robo_cylinder/src/nodes/robo_cylinder_services.py

```diff
@@ -96,139 +96,94 @@ def power(req):
 
 # Homing signal
 def home(req):
-    incorrect = 0
-    while incorrect < 5:
+    init_pos = pos_meters # initial position of car is defined here.
+    if init_pos <= 0.005: #  if the position is already home, it is true already. no need to proceed
```

Comment:
in general, bad to hard code values in the function. This comment applies to many lines of code in the file.